Backpack/StarRating/Classes/BPKHotelStarRating.swift

```diff
@@ -0,0 +1,80 @@
+//
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright © 2023 Skyscanner Ltd. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import UIKit
+
+public class BPKHotelStarRating: UIStackView {
+    
+    public var rating: Int {
+        didSet {
+            guard rating != oldValue else { return }
+            updateLookAndFeel()
+        }
+    }
+    
+    public var size: BPKStarSize {
+        didSet {
+            guard size != oldValue else { return }
+            updateLookAndFeel()
+        }
+    }
+    
+    public init(size: BPKStarSize = .small, rating: Int = 0) {
+        self.size = size
+        self.rating = rating
+        super.init(frame: .zero)
+        setup()
+    }
+    
+    public override init(frame: CGRect) {
+        self.size = .small
+        self.rating = 0
+        super.init(frame: frame)
+        setup()
+    }
+    
+    public required init(coder: NSCoder) {
+        self.size = .small
+        self.rating = 0
+        super.init(coder: coder)
+        setup()
+    }
+    
+    private func setup() {
+        translatesAutoresizingMaskIntoConstraints = false
+        axis = .horizontal
+        updateLookAndFeel()
+    }
+    
+    private func updateLookAndFeel() {
+        arrangedSubviews.forEach { view in
+            removeArrangedSubview(view)
+            view.removeFromSuperview()
```

Comment:
Do you need to call `removeFromSuperview()` after removing from the stackView? I think you dont need to do it, but please check!

if so: code could look like this
```
arrangedSubviews.forEach(removeArrangedSubview) // passing the method reference!
```swift

Reply:
This is needed, nevermind!

Reply:
As @gert-janvercauteren  mentioned earlier, we do need to call the removeArrangedSubview
Stackoverflow: https://stackoverflow.com/a/53260606
Apple's document: https://developer.apple.com/documentation/uikit/uistackview/1616235-removearrangedsubview